analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/RestrictDeserializedTypes.cs

```diff
@@ -35,6 +35,28 @@ public override bool ShouldExecute()
         return walker.Result;
     }
 
+    protected override SyntaxNode FindBindToTypeMethodDeclaration(IOperation operation) =>
+        MethodCandidates(operation)?.FirstOrDefault(x =>
```

Comment:
```suggestion
        MethodCandidates(operation).FirstOrDefault(x =>
```
Assigning `null` to the `Binder` property is "Compliant". Therefore it is checked in the rule implementation and never passed to `FindBindToTypeMethodDeclaration`. Thus `MethodCandidates` can never return `null` here.
This also increases code coverage.

Reply:
It throws an `ArgumentNullException` if remove it. `MethodCandidates` returns `null` if a `null` value is passed to the JavascriptSerializer as resolver.